src/test/llmnode_test.cpp

```diff
@@ -2652,16 +2652,128 @@ class EmbeddingsHttpTest : public ::testing::Test {
 };
 std::unique_ptr<std::thread> EmbeddingsHttpTest::t;
 
-TEST_F(EmbeddingsHttpTest, invalidModel) {
+const int EMBEDDING_OUTPUT_SIZE = 384;
+
+TEST_F(EmbeddingsHttpTest, simplePositive) {
     std::string requestBody = R"(
         {
-            "model": "invalidModel",
+            "model": "embeddings",
             "input": "dummyInput"
         }
     )";
     ASSERT_EQ(
         handler->dispatchToProcessor(endpointEmbeddings, requestBody, &response, comp, responseComponents, &writer),
-        ovms::StatusCode::MEDIAPIPE_DEFINITION_NAME_MISSING);
+        ovms::StatusCode::OK);
+    rapidjson::Document d;
+    d.Parse(response.c_str());
+    ASSERT_EQ(d["object"], "list");
+    ASSERT_EQ(d["data"].IsArray(), true);
+    ASSERT_EQ(d["data"].Size(), 1);
+    ASSERT_EQ(d["data"][0]["object"], "embedding");
+    ASSERT_EQ(d["data"][0]["embedding"].IsArray(), true);
+    ASSERT_EQ(d["data"][0]["embedding"].Size(), EMBEDDING_OUTPUT_SIZE);
+    ASSERT_EQ(d["data"][0]["index"], 0);
+}
+
+TEST_F(EmbeddingsHttpTest, simplePositiveBase64) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": "dummyInput",
```

Comment:
what about negative tests?
missing tests related to:
- what happens when pipeline is available but one of the models did not load properly? have wrong input name? have wrong output name? have wrong shape? non matching inputs/outputs?
- what happens when we request the endpoint via KServe grpc interface
- what happens when the connection is dropped during processing

if its not scope of this task, please specify which JIRA ticket will deliver those tests

Reply:
Noted in jira